app/main/views/frameworks.py

```diff
@@ -61,13 +61,6 @@ def create_framework():
             lots=lots
         )
         db.session.add(framework)
-        db.session.add(
-            AuditEvent(
-                audit_type=AuditTypes.create_framework,
-                db_object=framework,
-                user=updater_json['updated_by'],
-                data={'update': json_payload['frameworks']})
-        )
```

Comment:
Rather than move it out of the transaction you can do (I think, but look it up) `db.session.flush()` which will get the ID you need but keep the things within a single transaction.

I'm pretty sure this is a thing we do elsewhere.

We definitely want the two things to be created in a single commit - so either both pass or both fail.  We don't want a situation where the data gets written but the audit event fails.

Reply:
See eg here: https://github.com/alphagov/digitalmarketplace-api/blob/5cb28313a05b510bb9df04379b93ec7849ba3942/app/main/views/suppliers.py#L163

Reply:
Ah nice. Did not know that was a thing. Will fix up.